src/test/kotlin/org/sd_jwt/SdJwtKtTest.kt

```diff
@@ -174,4 +178,402 @@ internal class SdJwtKtTest {
             verifyPresentation<IdCredential>(noneHeaderHolderBindingPresentation, trustedIssuers, nonce, verifier, true)
         }
     }
+
+    @Test
+    fun testAdvancedCredentialStructured_Json_ok() {
```

Comment:
What do you think about putting these test cases in a separate file? I think this would make it cleaner.